app/src/main/java/com/iyehuda/feelslike/ui/map/MapFragment.kt

```diff
@@ -1,37 +1,222 @@
+            Post("2", LatLng(34.0522, -118.2437), "LA Post", "Content 2"),
+            Post("3", LatLng(51.5074, -0.1278), "London Post", "Content 3"),
+            Post("4", LatLng(35.6762, 139.6503), "Tokyo Post", "Content 4"),
+            Post("5", LatLng(32.0853, 34.7818), "Tel Aviv Post", "Welcome to Tel Aviv!")
+        ))
+    }
+
+    override fun onMapReady(googleMap: GoogleMap) {
+        map = googleMap
+        enableMyLocation()
+        displayPosts()
+        // Center on Tel Aviv with zoom level 12
+        map.moveCamera(CameraUpdateFactory.newLatLngZoom(LatLng(32.0853, 34.7818), 12f))
+    }
+
+    private fun displayPosts() {
+        posts.forEach { post ->
+            // Create custom marker
+            addCustomMarker(
+                latLng = post.location,
+                feelsLike = "Feels Hot",
+                temperature = "24°C"
+            )
+
+            // Add click listener
+            map.setOnMarkerClickListener { clickedMarker ->
+                // Show post details
+                val clickedPost = posts.find { it.location == clickedMarker.position }
+                clickedPost?.let { showPostDetails(it) }
+                true
+            }
+        }
+    }
+
+    private fun showPostDetails(post: Post) {
+        // Show a bottom sheet or dialog with post details
+        val bottomSheet = BottomSheetDialog(requireContext())
+        val view = LayoutInflater.from(requireContext()).inflate(R.layout.post_details_bottom_sheet, null)
+
+        view.findViewById<TextView>(R.id.titleText).text = post.title
+        view.findViewById<TextView>(R.id.contentText).text = post.content
+
+        bottomSheet.setContentView(view)
+        bottomSheet.show()
+    }
+
+    private fun enableMyLocation() {
+        if (ContextCompat.checkSelfPermission(
+                requireContext(),
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ) == PackageManager.PERMISSION_GRANTED
+        ) {
+            map.isMyLocationEnabled = true
+        } else {
+            // Request location permission
+            ActivityCompat.requestPermissions(
+                requireActivity(),
+                arrayOf(Manifest.permission.ACCESS_FINE_LOCATION),
+                LOCATION_PERMISSION_REQUEST_CODE
+            )
+        }
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<String>,
+        grantResults: IntArray
+    ) {
+        if (requestCode == LOCATION_PERMISSION_REQUEST_CODE) {
+            if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                enableMyLocation()
+            }
+        }
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+    }
+
+    private fun addCustomMarker(
+        latLng: LatLng, 
+        feelsLike: String, 
+        temperature: String,
+        profileImageResource: Int? = null
```

Comment:
The image should be a `Uri` object since the images are fetched from Firebase.
Use `ImageUtil` to load the image